akka-actor/src/main/scala/akka/dispatch/Mailbox.scala

```diff
@@ -862,7 +862,7 @@ final case class BoundedControlAwareMailbox(capacity: Int, override final val pu
 }
 
 object BoundedControlAwareMailbox {
-  class MessageQueue(val capacity: Int, val pushTimeOut: FiniteDuration) extends BoundedControlAwareMessageQueueSemantics {
+  class MessageQueue(val capacity: Int, val pushTimeOut: FiniteDuration) extends BoundedControlAwareMessageQueueSemantics with java.io.Serializable {
```

Comment:
The tests send `MessageQueue` as messages, so they fail if the `MessageQueue` is not Serializable. It felt odd to me, but as far as I could track all other `MessageQueues` are also Serializable (by inheritance, e.g. `BoundedDequeBasedMailbox.MessageQueue` extends `LinkedBlockingDeque`, which is Serializable). Also, it makes sense if we consider queues store Messages, which are Serializable as well.

Therefore, for consistency sake, I decided to make `BoundedControlAwareMailbox.MessageQueue` Serializable. However, I must say that I haven't thought about binary compatibility here. If needed, I can remove the extends and create some other tests for this functionality